examples/ec2_remote_proxy/index.ts

```diff
@@ -0,0 +1,96 @@
+import { interpolate, Config, secret } from "@pulumi/pulumi";
+import { local, remote, types } from "@pulumi/command";
+import * as aws from "@pulumi/aws";
+import * as fs from "fs";
+import * as os from "os";
+import * as path from "path";
+import { size } from "./size";
+
+const config = new Config();
+const keyName = config.get("keyName") ?? new aws.ec2.KeyPair("key", { publicKey: config.require("publicKey") }).keyName;
+const privateKeyBase64 = config.get("privateKeyBase64");
+const privateKey = privateKeyBase64 ? Buffer.from(privateKeyBase64, 'base64').toString('ascii') : fs.readFileSync(path.join(os.homedir(), ".ssh", "id_rsa")).toString("utf8");
+
+const secgrp = new aws.ec2.SecurityGroup("secgrp", {
+    description: "Foo",
+    ingress: [
+        { protocol: "tcp", fromPort: 22, toPort: 22, cidrBlocks: ["0.0.0.0/0"] },
+        { protocol: "tcp", fromPort: 80, toPort: 80, cidrBlocks: ["0.0.0.0/0"] },
+    ],
+    egress: [
+        { fromPort: 0, toPort: 0, protocol: "-1", cidrBlocks: ["0.0.0.0/0"], ipv6CidrBlocks: ["::/0"], },
+    ],
+});
+
+const ami = aws.ec2.getAmiOutput({
+    owners: ["amazon"],
+    mostRecent: true,
+    filters: [{
+        name: "name",
+        values: ["amzn-ami-hvm-*-x86_64-gp2"],
+    }],
+});
+
+const server = new aws.ec2.Instance("server", {
+    instanceType: size,
+    ami: ami.id,
+    keyName: keyName,
+    vpcSecurityGroupIds: [secgrp.id],
```

Comment:
Maybe not strictly necessary - but you could use a separate security group that restricts ingress to the proxy servers security group, so that access is not possible directly and must go through the proxy to really prove that this is working (and not just skipping the proxy).

Reply:
I like that idea. Especially since its the "examples" folder, and that is a much more compelling example.